ejmask-extensions/src/main/java/com/ebay/ejmask/extenstion/builder/json/JsonBooleanFieldPatternBuilder.java

```diff
@@ -0,0 +1,108 @@
+package com.ebay.ejmask.extenstion.builder.json;
+/**
+ * Copyright (c) 2023 eBay Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import com.ebay.ejmask.api.PatternEntity;
+import com.ebay.ejmask.extenstion.builder.AbstractRegexPatternBuilder;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.ebay.ejmask.core.util.CommonUtils.emptyIfNull;
+
+/**
+ * An implementation of IPatternBuilder to support sensitive JSON field, whose value need to be partially masked.
+ * This builder is for masking the field value with a Boolean type.
+ *
+ * @author fsun1
+ */
+public class JsonBooleanFieldPatternBuilder extends AbstractRegexPatternBuilder {
+
+    private static final List<PatternEntity> PATTERN_ENTITY_LIST = Arrays.asList(
+            /**
+             * Boolean field with value to be masked
+             * @link https://regex101.com/r/AEwc99/1
+             */
+            new PatternEntity("\\\"(%s)(\\\\*\\\"\\s*:\\s*\\\\*)(\\b(true|TRUE|True|false|FALSE|False)\\b)([^\\\"]{1,3})[^\\\"]*(\\\\?\\\"|)", "\"$1$2\"xxxx\"$5$6")
```

Comment:
What is the usecase of masking boolean?

Reply:
just like string or numeric data, as a framework, ejmask needs to provide the ability to mask commonly used data types.

Reply:
Keeping this as this is harmless.